fmc/src/flow/rt_alias.rs

```diff
@@ -11,40 +11,168 @@ Abstract:
     Alias RT DICE Layer & PCR extension
 
 --*/
+use crate::flow::crypto::Crypto;
 use crate::flow::dice::{DiceInput, DiceLayer, DiceOutput};
 use crate::flow::pcr::{extend_current_pcr, extend_journey_pcr};
+use crate::flow::tci::Tci;
+use crate::flow::x509::X509;
 use crate::fmc_env::FmcEnv;
 use crate::HandOff;
 use caliptra_common::cprintln;
-use caliptra_drivers::{CaliptraError, CaliptraResult};
+use caliptra_common::crypto::Ecc384KeyPair;
+use caliptra_drivers::{
+    okref, CaliptraError, CaliptraResult, Ecc384PubKey, Hmac384Data, Hmac384Key, KeyId, KeyReadArgs,
+};
+const SHA384_HASH_SIZE: usize = 48;
 
 #[derive(Default)]
 pub struct RtAliasLayer {}
 
 impl DiceLayer for RtAliasLayer {
     /// Perform derivations for the DICE layer
-    fn derive(_env: &FmcEnv, _input: &DiceInput) -> CaliptraResult<DiceOutput> {
-        // TODO : implement derivation.
-        Err(CaliptraError::FMC_RT_ALIAS_UNIMPLEMENTED)
+    fn derive(
+        env: &mut FmcEnv,
+        hand_off: &HandOff,
+        input: &DiceInput,
+    ) -> CaliptraResult<DiceOutput> {
+        // Derive CDI
+        let cdi = *okref(&Self::derive_cdi(env, hand_off, input.cdi))?;
+
+        // Derive DICE Key Pair from CDI
+        let key_pair = Self::derive_key_pair(env, cdi, input.subj_priv_key)?;
+
+        // Generate the Subject Serial Number and Subject Key Identifier.
+        //
+        // This information will be used by next DICE Layer while generating
+        // certificates
+        let subj_sn = X509::subj_sn(env, &key_pair.pub_key)?;
+        let subj_key_id = X509::subj_key_id(env, &key_pair.pub_key)?;
+
+        let output = input.to_output(key_pair, subj_sn, subj_key_id);
+
+        // Generate Local Device ID Certificate
+        Self::generate_cert_sig(env, input, &output)?;
+        Ok(output)
     }
 }
 
 impl RtAliasLayer {
     #[inline(never)]
     pub fn run(env: &mut FmcEnv, hand_off: &HandOff) -> CaliptraResult<()> {
-        cprintln!("[art] Extend PCRs");
+        cprintln!("Extend PCRs");
```

Comment:
Why was [art] removed? Think it would be good to have a tag about what stage this is. Maybe `[fmc] Extend RT PCRs`?

Reply:
done